Fix relative reference resolution within bundles

abjonnes opened the pull request:
Fixes relative reference resolution for references within a bundle by moving the server reference from abstract resource model to the abstract base model, and walking up the owner hierarchy directly rather than using `owningResource` to walk up the chain of domain resources (which doesn't include `Bundle`).

Also renames the `server` property to `origin_server` to reduce potential name conflicts with FHIR resources.

p2 commented:
For semantic reasons I'd prefer if the base model didn't have a hold on server since it's not possible for an element to come back from a server, just for a resource. Do you think you can achieve the same by leveraging `_owner`/`owningResource` instead?

abjonnes commented:
Thanks for the feedback. I moved the server reference back to the resource model and just changed the lookup to iteratively move up the owner chain via `_owner` until a server is found.
